src/redux/actions/order.js

```diff
@@ -23,6 +28,51 @@ const add = () => {
   }
 }
 
+const txStatusCheckAndUpdateOrder = order => {
+  return async (dispatch, getState, { getFirestore }) => {
+    const firestore = getFirestore()
+
+    if (!order.txConfirmed && order.txHash) {
+      try {
+        const status = await web3Instance.isTxConfirmed(order.txHash)
+        await firestore
+          .collection('orders')
+          .doc(order.id)
+          .update({
+            txConfirmed: status,
+          })
+        if (status) {
+          // updating selling items
+          const confirmed_order = await firestore
+            .collection('orders')
+            .doc(order.id)
+            .get()
+          const items = confirmed_order.data().items
+          _.map(items, async (item, id) => {
+            const fb_item = await firestore
+              .collection('items')
+              .doc(id)
+              .get()
+
+            await firestore
+              .collection('items')
+              .doc(id)
+              .update({
+                quantity: fb_item.data().quantity - item.sellingQuantity,
+                soldCount: fb_item.data().soldCount + item.sellingQuantity,
+              })
+          })
+        }
+        dispatch({ type: ACTIONS.UPDATE_ORDER })
```

Comment:
Hm... `firebase-redux` will update the state for us. 

Don't think we need to dispatch anything if we are making a firebase update unless we also want to store something else in Redux state.